pkg/readiness/ready_tracker.go

```diff
@@ -333,6 +345,31 @@ func (t *Tracker) collectInvalidExpectations(ctx context.Context) {
 	}
 }
 
+func (t *Tracker) trackAssignMetadata(ctx context.Context) error {
+	if !*mutation.MutationEnabled {
+		return nil
+	}
+	defer func() {
```

Comment:
I think this defer needs to go above the flag switch.

IIRC if `ExpectationsDone()` is never set, `Satisfied()` never returns true.

Reply:
+1